### PR TITLE
Move code to `main()`

### LINKED ISSUE #2: Move code into main() function
In order to be able to turn this into a Python module that's importable as a library, or runnable as CLI application, we need to move the "global" code into a main function like this:

```python
def main():
    ...

if __name__ == '__main__':
     main()
```

### PR DESCRIPTION
**Pull request by danielhollas**
This is a last PR that mostly just moves code. Now that all the code is out of global scope, we can make this into an importable module / package in next PR. 

This also exposed a couple of cases in `sample_initial_conditions()` method which relied on global variables.

Closes #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,8 +6,6 @@ In the future, the authors plan to turn the script into a Python library that ca
 
 ### Necessary python libraries
 The code was tested with Python 3.7 and higher. The necessary python libraries and their version that were used in development:
-* `argparse`
```
Comment (danielhollas) [inline README.md:9]: These two modules are in standard library, so are always available, no need to mention them here.

File: promdens.py [hunk 2]
```diff
@@ -689,10 +696,22 @@ def plot_pda(ics: InitialConditions) -> None:
...
+        ics.ics[3] / ics.evtoau,
+        ics.ics[1] / ics.fstoau,
+        range=[[emin, emax], [tmin, tmax]],
+        bins=(100, 100),
+        cmap=plt.cm.viridis,
+        density=True,
+    )
+    if ics.field_lchirp != 0:
+        axs.plot(
+            (ics.field_omega + 2 * ics.field_lchirp * ics.field_t) / ics.evtoau,
```
Comment (danielhollas) [inline promdens.py:709]: Here I've switched from `omega` to `ics.field_omega` and `lchirp` to `ics.field_lchirp`.

**Review by JanosJiri — CHANGES_REQUESTED** [review]
Very nice, thanks for all the work! I located one or two typos, otherwise I see no problems with the code. I did a simple test and it worked.

I have only a small issue with some changes in formatting of a few lines of the code. I would keep input parameters of functions as one line of the code unless you have some objections against it. I use a autoformatting in PyCharm before every commit and it changes it back to the original one line structure. I would run it through the autoformatting in the future, which would remove that. Is it ok for you?

**Comment by danielhollas** [comment]
@JanosJiri thanks for the review. Agree that formatting is a matter of taste and should be dealt separately. Ideally in the future we'd have autoformatter running automatically in CI so that the formatting doesn't depend on personal configurations. 

Personally I find that a lot of the lines in the current code are quite long. But again that's a matter of preference.

For now could you please autoformat the file according to your config and push to this branch. Feel free to fix the typos as well.

**Comment by JanosJiri** [comment]
@danielhollas I see your point. Maybe it would be best to later find a way in between with the formatting. We can later set up a format to solve it. Thanks again for the work. I have changed the typos and I'm going to squash and merge the pull request.

**Review by JanosJiri — APPROVED** [review]

**Pull request merged**